src/lib/docker/dockerService.ts

```diff
@@ -387,6 +398,151 @@ class DockerService implements DockerLibrary {
       }
     }
   }
+
+  /**
+   * Constructs the contents of sim.json file for the simulation
+   *
+   * @param network the network to start
+   */
+  constructSimJson(network: Network) {
+    // Helper function to convert Windows paths to POSIX path format.
+    const getPosixPath = (path: string) => {
+      // Normalize to POSIX separators for Windows paths.
+      const norm = path.replace(/\\/g, '/');
+
+      const parts = norm.split('volumes/');
+
+      return parts[parts.length - 1];
+    };
+    const simJson: {
+      nodes: SimulationNodeConfig[];
+      activity: ActivityConfig[];
+    } = {
+      nodes: [],
+      activity: [],
+    };
+
+    const simulation = network.simulation;
+    if (!simulation) return { nodes: [], activity: [] };
+
+    const { activity } = simulation;
+    const { lightning } = network.nodes;
+
+    activity.forEach(a => {
+      const { source, destination, intervalSecs, amountMsat } = a;
+      const nodeNames = [source, destination];
+
+      for (const nodeName of nodeNames) {
+        let simNode: SimulationNodeConfig;
+
+        const node = lightning.find(n => n.name === nodeName);
+        if (!node) {
+          throw new Error(`Node ${nodeName} not found in network`);
+        }
+
+        // Split the macaroon and cert path at "volumes/" to get the relative path
+        // to the docker volume. This is necessary because the docker volumes are
+        // mounted as a different path in the container.
+        switch (node.implementation) {
+          case 'LND':
+            const lnd = node as LndNode;
+            simNode = {
+              id: lnd.name,
+              macaroon: `/home/simln/.${getPosixPath(lnd.paths.adminMacaroon)}`,
+              address: `https://host.docker.internal:${lnd.ports.grpc}`,
```

Comment:
Using `host.docker.internal` doesn't work on Linux. It gives me the following error when the `simln` node starts up:
```
$ docker logs -f polar-n1-simln
Error: Get info error: status: Unavailable, message: "error trying to connect: dns error: failed to lookup address information: Name does not resolve", details: [], metadata: MetadataMap { headers: {} }
```

I manually updated my `simln.json` file to use `https://polar-n1-alice:10009` and that worked on Linux. I still need to test this on the other platforms as well.